package/MDAnalysis/core/selection.py

```diff
@@ -838,9 +847,11 @@
         return group[mask].unique
 
 
-class ResnumSelection(RangeSelection):
-    token = 'resnum'
-    field = 'resnums'
+class FloatRangeSelection(RangeSelection):
+    """Range selection for float values"""
+
+    pattern = f"({FLOAT_PATTERN}){RANGE_PATTERN}({FLOAT_PATTERN})"
+    dtype = float
```

Comment:
Maybe for matching floats we should be using a fuzzier match, something like np.almost_equal? Or would that just create bad results for positions?

Reply:
I don't really want to encourage equality comparisons for floats. IIRC there's already a warning in the docs that for floats one should do range comparisons.

But if you think that we get less grief by having some "almost equal" matching for floats then we could do it. 
What would we choose as tolerance? What algorithm for comparison (using only absolute, only relative, or whatever [np.allclose()](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html) does where we set the default values?

Or we could introduce a `~` operator for "almost equal matching". Or to be very onerous, the unicode symbol [U+2248 ALMOST EQUAL TO](http://www.unicode-symbol.com/u/2248.html) `≈`.

Reply:
> What would we choose as tolerance?

I made this the user's problem by adding ``rtol`` and ``atol`` keywords. I don't know how to get around things like the below though (or if we should be trying):

```python
>>> 0.3000000000000000000000000 == 0.30000000000000001
True
>>> 0.3000000000000000000000000 is 0.30000000000000001
True
>>> 0.30000000000000001 == 3/10
True
>>> 0.30000000000000001 is 3/10
True
```

Reply:
This also brings up what to do with the PropertySelection ("prop mass == 0.3") and SameSelection ("same mass as (mass 0.3)"). For now I changed the former to use `np.isclose` but the latter checks for value membership with equality (`np.in1d`)